src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -98,7 +99,59 @@ public function connect()
             }
         }
     }
-    
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasSchemaTable()
+    {
+        if (! $this->hasTable($this->getSchemaTableName())) {
+            return false;
+        }
+
+        if (! $this->hasIndex($this->getSchemaTableName(), 'version')) {
```

Comment:
the original phinx schema has no keys so this is how we detect it.
This is also why we `GROUP BY version` when inserting, to remove duplicates which may have occurred